### PR TITLE
continue hashing if data is none

### LINKED ISSUE #14907: load_package failed if there is empty file in package
### What is the problem?
When testing the experimental load_package, the function could fail if there is an empty file in the package folder, especially if the empty file is read in first during the hashing loop in runtime_env's '_hash_modules'

*Ray version and other system information (Python version, TensorFlow version, OS):*
* Ray version: master branch
* Python 3.8.3
* OS: Darwin Kernel Version 19.6.0
### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

```
import ray
from ray.experimental.packaging.load_package import load_package
op = load_package("./python/ray/experimental/packaging/example_pkg/ray_pkg.yaml")
```
The above code will works fine. but if in example_pkg, adding an empty file, e.g., \_\_init\_\_.py, 
The code will fail.

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by valiantljk-tt**
## Why are these changes needed?
In runtime_env, the function '_hash_modules' will go through all files in a given directory and read the first few bytes to generate a hash value. However, in case of an empty file, the current code will 'break' the for loop. 
This looping should 'continue' to loop through all files. 

## Related issue number
Closes #14907

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by valiantljk-tt** [comment]
@ericl

File: python/ray/_private/runtime_env.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def _hash_modules(path: Path) -> bytes:
             with open(from_file_name, mode="rb") as f:
                 data = f.read(BUF_SIZE)
```
Comment (ericl) [inline python/ray/_private/runtime_env.py:142]: Hmm @iycheng isn't this only including the first BUF_SIZE bytes from each file then?

**Review by ericl — APPROVED** [review]

File: python/ray/_private/runtime_env.py (hunk 1 above)
Comment (iycheng) [reply python/ray/_private/runtime_env.py:142]: Yes, I feel I forgot the loop here.  It should be:

```
data = f.read(BUF_SIZE)
while data is not None:
    md5.update(data)
    data = f.read(BUF_SIZE)
```

@valiantljk-tt thanks for the fix. Could you update it to something like this?

Comment (valiantljk-tt) [reply python/ray/_private/runtime_env.py:142]: sure

Comment (valiantljk-tt) [reply python/ray/_private/runtime_env.py:142]: one question, do we need to read all data from the file to produce the hash? what if the file is very large?
I thought, the purpose of this is generate a unique name, so first BUF_SIZE bytes from each file should be fine, right?
@iycheng

Comment (iycheng) [reply python/ray/_private/runtime_env.py:142]: We need a better algorithm for this one in the future. Right now, to make it safe, we calculate caching for all files. Later, we can cache the hash value of these files locally and only recalculate them when the data changed.

Comment (valiantljk-tt) [reply python/ray/_private/runtime_env.py:142]: ok, that makes sense, let me submit the change.

Comment (valiantljk-tt) [reply python/ray/_private/runtime_env.py:142]: @iycheng just submitted, plz review. thx

**Comment by ericl** [comment]
//python/ray/tests:test_runtime_env                                     TIMEOUT in 3 out of 3 in 915.0s
  | Stats over 3 runs: max = 915.0s, min = 915.0s, avg = 915.0s, dev = 0.0s
  | /root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/bazel-out/k8-opt/testlogs/python/ray/tests/test_runtime_env/test.log
  | /root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/bazel-out/k8-opt/testlogs/python/ray/tests/test_runtime_env/test_attempts/attempt_1.log
  | /root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/bazel-out/k8-opt/testlogs/python/ray/tests/test_runtime_env/test_attempts/attempt_2.log

**Review by fishbone — APPROVED** [review]

**Comment by ericl** [comment]
@iycheng @valiantljk-tt could you look at the test timeout? It seems directly related.

**Comment by valiantljk-tt** [comment]
I'm trying to test locally, but I'm not able to build ray from source, #14752

**Comment by iycheng** [comment]
@valiantljk-tt we recently fixed some bugs related with runtime env, could you merge to master and give it another try?

**Comment by valiantljk-tt** [comment]
> @valiantljk-tt we recently fixed some bugs related with runtime env, could you merge to master and give it another try?

okey

File: python/ray/_private/runtime_env.py [hunk 2]
```diff
@@ -139,9 +139,9 @@ def _hash_modules(path: Path) -> bytes:
         if not Path(from_file_name).is_dir():
             with open(from_file_name, mode="rb") as f:
                 data = f.read(BUF_SIZE)
-                if not data:
-                    break
-                md5.update(data)
+                while data is not None:
```
Comment (iycheng) [inline python/ray/_private/runtime_env.py:142]:
```suggestion
                while len(data) != 0:
```

**Pull request merged**